### ISSUE TITLE
animating rotations -180 t0 180 without baking causes extreme model sheer. 

### ISSUE THREAD

**Issue by spiraloid** [KhronosGroup/glTF-Blender-IO#1253]
**Describe the bug**
try to set keys and export rotations that are 180 or 360 degrees apart without baking the animation on every frame.  the model will scale.

**To Reproduce**
create monkey.
go to frame 0
keyframe xyz rotations.
go to frame 72
type 360 into z rotation
keyframe xyz rotations.
export gltf file.
see the model scale to 0.

**Expected behavior**
the rotating model should not scale.  it should rotate 360 degrees.  it should not need to bloat the file by baking every keyframe.

**Screenshots**
https://youtu.be/hY9UNKwo18Y?t=1599

**.blend file/ .gltf**
If applicable, a zipped folder containing a .blend file for exporter issues and a .gltf file for importer issues. Please take care to include all resources, such as textures in the .zip file.

**Version**
 - OS: [e.g. macOS, linux]
 - Blender Version [e.g. 2.82, 2.83, or build date]

**Additional context**

**Comment by julienduroure** [comment]
Hello,
What is your gltf reader?
I can't reproduce either with https://gltf-viewer.donmccurdy.com/ or re-importing in blender.
Can you please upload your generated glb file?
Thanks

**Comment by donmccurdy** [comment]
I'm able to reproduce the issue, here's a sample `.blend` and exports with and without "Always sample animations" enabled:

[Spin.zip](https://github.com/KhronosGroup/glTF-Blender-IO/files/5442129/Spin.zip)

Rotations in glTF are stored as quaternions, and thus a 360º rotation cannot be represented with just two keyframes. So you'll get the correct result when "Always sample animations" is enabled, but not otherwise. The only way the exporter will be able to avoid this issue would be for it to add enough keyframes that the distance between keyframes is <180º, when it detects this issue. I'm not sure how straightforward it would be for the exporter to insert additional keyframes in these cases.

A few workarounds are possible in the meantime:

- A. Add 1-2 keyframes in between manually, so that the distance between keyframes is <180º.
- B. Export with "Always sample animations" enabled, and then optimize the keyframes out afterward, using [gltfpack](https://github.com/zeux/meshoptimizer).

**Comment by julienduroure** [comment]
Ooops, I forgot that Sampled animation is now ON by default. So, yes, I can reproduce.

Agreed with Don analysis

**Comment by spiraloid** [comment]
I wrote an auto bake and simplify curve on export for my http://3dcomic.shop gltf comics.

my exporter can be found here if you want to see it:
https://github.com/spiraloid/Spiraloid-Toolkit-for-Blender-3DComicToolkit

IMHO, baking/resampling bloats the .glb file sizes so I'd prefer the interpolation function in the same way as in blender if possible.

**Comment by donmccurdy** [comment]
> I wrote my own export preprocessor to combine, bake, covert grease penci and simplify anims curves for my 3dcomics.

Oh that's awesome! Really nice look to those. :) 

> IMHO, baking/resampling bloats the .glb file sizes so I'd prefer the interpolation function in the same way as in blender if possible.

This is only sometimes possible, and (at least to my understanding) even detecting all of the cases where resampling is needed can be a hard problem of its own... I think we might have better luck trying to trim redundant keyframes after resampling animation data honestly.

**Comment by spiraloid** [comment]
yeah that was my conclusion too. if you look at the next version of my 3D Comic toolkit export_panel function,  I’m actually doing this step already.  Blenders 2.91+ fcurve decimation-by-error works pretty nicely.
